src/main/java/org/jenkinsci/plugins/junitrealtimetestreporter/PipelineAttacher.java

```diff
@@ -0,0 +1,211 @@
+    @Override
+    public void onNewHead(FlowNode node) {
+        if (LOGGER.isLoggable(Level.FINER)) {
+            LOGGER.log(Level.FINER, "notified of {0} ~ {1}", new Object[] {node, node.getDisplayFunctionName()});
+        }
+        if (node instanceof BlockStartNode) {
+            for (FlowNode node2 : node.getParents()) { // do not pay attention until the body is invoked, when WorkspaceAction is attached
+                LOGGER.log(Level.FINER, "considering of {0} ~ {1}", new Object[] {node2, node2.getDisplayFunctionName()});
+                WorkspaceAction wsa = node2.getPersistentAction(WorkspaceAction.class);
+                if (wsa != null) {
+                    FilePath workspace = wsa.getWorkspace();
+                    if (workspace == null) {
+                        LOGGER.fine("no workspace");
+                        return;
+                    }
+                    Queue.Executable executable;
+                    try {
+                        executable = node2.getExecution().getOwner().getExecutable();
+                    } catch (IOException x) {
+                        LOGGER.log(Level.WARNING, null, x);
+                        return;
+                    }
+                    if (executable instanceof Run) {
+                        Run<?, ?> run = (Run<?, ?>) executable;
+                        Job<?, ?> job = run.getParent();
+                        if (!PerJobConfiguration.isActive(job)) {
+                            LOGGER.log(Level.FINE, "inactive in {0}", job);
+                            return;
+                        }
+                        LOGGER.log(Level.FINE, "encountered {0} in {1} entering {2}", new Object[] {workspace, run, node2.getDisplayFunctionName()});
+                        Run<?, ?> last = job.getLastSuccessfulBuild();
+                        if (!(last instanceof FlowExecutionOwner.Executable)) {
+                            LOGGER.log(Level.FINE, "no lastSuccessfulBuild in {0}", job);
+                            return;
+                        }
+                        FlowExecutionOwner owner = ((FlowExecutionOwner.Executable) last).asFlowExecutionOwner();
+                        if (owner == null) {
+                            LOGGER.log(Level.WARNING, "could not get FlowExecutionOwner from {0}", last);
+                            return;
+                        }
+                        FlowExecution exec = owner.getOrNull();
+                        if (exec == null) {
+                            LOGGER.log(Level.WARNING, "could not get FlowExecution from {0}", owner);
+                            return;
+                        }
+                        boolean keepLongStdio = false;
+                        Set<String> globs = new TreeSet<String>();
+                        for (FlowNode n : new DepthFirstScanner().allNodes(exec)) {
+                            if (n instanceof StepNode && ((StepNode) n).getDescriptor().getFunctionName().equals("step")) {
```

Comment:
So this would need to get tweaked post https://github.com/jenkinsci/junit-plugin/pull/76 to look for the `junit` step as well? I feel like trying to do any more granular stage-to-stage comparison may be a mess, though - depends on too much correlation of structure from one build to the next, which we can't guarantee or really expect...hmm...

Reply:
> to look for the `junit` step as well?

`junitResults` that is: https://github.com/jenkinsci/junit-plugin/pull/76#pullrequestreview-60776207

> too much correlation of structure from one build to the next

Well, we can expect `stage` names align in most cases (and just suppress this feature in whole or in part when they do not).